src/cachelib/redis.py

```diff
@@ -57,6 +57,11 @@ def __init__(
             self._read_client = self._write_client = host
         self.key_prefix = key_prefix or ""
 
+    def _get_prefix(self):
+        return (
+            self.key_prefix if isinstance(self.key_prefix, str) else self.key_prefix()
```

Comment:
Not really an issue here, but I wonder if using On a side note, I wonder if using https://docs.python.org/3/library/functions.html#callable wouldn't be make more sense semantically. In the end we want to call the key if it's callable and use it as is if it's not